tests/lib/rules/no-node-access.test.ts

```diff
@@ -51,18 +47,16 @@ ruleTester.run(RULE_NAME, rule, {
         within(signinModal).getByPlaceholderText('Username');
       `,
     },
-    /*{
-      // TODO: this one should be valid indeed. Rule implementation must be improved
-      //  to track where the nodes are coming from. This one wasn't reported before
-      //  just because this code is not importing TL module, but that's a really
-      //  brittle check. Instead, this one shouldn't be reported since `children`
-      //  it's just a property not related to a node
+    {
       code: `
         const Component = props => {
           return <div>{props.children}</div>
         }
       `,
```

Comment:
This test is reenabled now, but just because the file name doesn't match so there is nothing reported here. As mentioned in my original comment there, the rule implementation is really brittle and should make sure node methods are coming from an actual node and not a var or prop. That's out of the scope of this refactor so I'm not gonna fix it for now (it shouldn't happen that often with file-name setting tho).